File: ordenamento/insertion_sort.py
def InsertionSort(lista):
    for index in range(0, len(lista)):
        while index > 0:
            if lista[index-1] > lista[index]:
                lista[index], lista[index-1] = lista[index-1], lista[index]
            index -= 1
            print(lista)
    return lista

numeros = [4,7,2,8,5,11,0,-1,7]
print(InsertionSort(numeros))

File: javascript/insertionsort.js
function InsertionSort(lista){
    const index_maximo = lista.length;

    for (i = 0; i < index_maximo; i++){
        let j = i;
        while (j > 0){
            if (lista[j-1] > lista[j]){
                [lista[j-1], lista[j]] = [lista[j], lista[j-1]];
            }
            j--;
        }
    }
    return lista;
}

let numeros = [7, 5, 2, 0, 4, 9, 3];

let result = InsertionSort(numeros);
console.log("Resultado");
console.log(result);
